app/controllers/moirai/translation_files_controller.rb

```diff
@@ -90,15 +91,27 @@ def redirect_to_translation_file(file_path)
 
     def set_translation_file
       @file_path = @file_handler.file_hashes[params[:id]]
+      if @file_path.nil?
+        flash.alert = "File not found"
+        redirect_to moirai_translation_files_path, status: :not_found
+        return
+      end
       @decoded_path = CGI.unescape(@file_path)
     end
 
     def translation_params
-      params.require(:translation).permit(:key, :locale, :value, :file_path)
+      params.require(:translation).permit(:key, :locale, :value)
     end
 
     def load_file_handler
       @file_handler = Moirai::TranslationFileHandler.new
     end
+
+    def translation_same_as_in_file?(key, value, file_path)
+      return false if file_path.blank?
+      return false unless File.exist?(file_path)
+
+      value == @file_handler.parse_file(file_path)[key]
+    end
```

Comment:
it seems to me that you don't really need any parameter for this method since everything is already available to you.

```
def translation_same_as_in_file?
  file_path = KeyFinder.new.file_path_for(translation_params[:key], locale: translation_params[:locale])
  return false if file_path.blank?
  return false unless File.exist?(file_path)
  translation_params[:value] == @file_handler.parse_file(file_path)[translation_params[:key]]
end
```